### ISSUE TITLE
cp from server with empty files does not work

### ISSUE THREAD

**Issue by equinox0815** [minio/mc#1876]
Hi,

I have a folder with a number of files, some of which have 0B size, on a minio server. Using `mc` i try to download all the files within that directory

## Expected behaviour

Downloading of all files should succeed. Even if the file has 0B size.

## Actual behaviour

All files which have a size of 0Bytes won't be downloaded.

```
mc: <ERROR> Failed to copy ‘http://10.11.2.15:9000/logstash/queue/logstash.2016-11-08-13_13_00.xz’. The requested range is not satisfiable
mc: <ERROR> Session safely terminated. To resume session ‘mc session resume mjakWTwC’
```

## Steps to reproduce the behaviour

 - upload a 0 byte file using `mc cp`
 - try to download the file using `mc cp` again.

## mc version
- (paste output of `mc version`)

```
Version: 2016-10-07T01:56:22Z
Release-tag: RELEASE.2016-10-07T01-56-22Z
Commit-id: 75d55783c01c9849055f76cca146390d2d5df631
```

## System information

Debian Jessie (amd64) using `mc` release binaries from mini.io.

regards
 christian

**Comment by krishnasrinivas** [comment]
@equinox0815 thanks for reporting.

`mc` is adding Range header:

```
mc: <DEBUG> GET /test/0b HTTP/1.10b                                                                     
Host: 127.0.0.1:9000
User-Agent: Minio (linux; amd64) minio-go/2.0.2 mc/DEVELOPMENT.GOGET
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**/20161108/us-east-1/s3/aws4_request, SignedHeaders=expect;host;range;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
Expect: 100-continue
Range: bytes=0-10485759
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20161108T125522Z
```

needs to be investigated.

**Comment by equinox0815** [comment]
Hi,

I had a little time to investigate a bit more and want to add that it might be a problem on the server-side. It is likely that the 0B file on the server was the result of a failed upload and the thus the file on the server was somehow corrupeted...
On the server side i was using minio (downloaded from minio.io 2 days ago).

Sorry i don't have time to investigate further...

regards
 christian

**Comment by vadmeste** [comment]
@equinox0815, thanks for your message.
 
I am currently investigating the issue. I see the problem with AWS S3 too, you may or not be experimenting a bug in Minio server. Anyway, I will update you soon.

**Comment by harshavardhana** [comment]
> I am currently investigating the issue. I see the problem with AWS S3 too, you may or not be experimenting a bug in Minio server. Anyway, I will update you soon.

The issue is in 'minio-go' i think @vadmeste

**Comment by vadmeste** [comment]
Fixed by #1879

**Comment by felixbuenemann** [comment]
This also affects `mc mirror remote local` with empty files on s3.

@vadmeste Will there be a new release with this fix soon?
